modules/objdetect/test/test_qrcode_encode.cpp

```diff
@@ -430,4 +430,80 @@ TEST(Objdetect_QRCode_Encode_Decode_Structured_Append, DISABLED_regression)
 
 #endif // UPDATE_QRCODE_TEST_DATA
 
+TEST(Objdetect_QRCode_Encode_Decode, regression_issue22029)
+{
+    const cv::String expect_msg = "OpenCV";
+    const int min_version = 1;
+    const int max_version = 40;
+
+    for ( int v = min_version ; v <= max_version ; v++ )
+    {
+        Mat qrimg;
+        QRCodeEncoder::Params params;
+        params.version = v;
+        Ptr<QRCodeEncoder> qrcode_enc = cv::QRCodeEncoder::create(params);
+        qrcode_enc->encode(expect_msg, qrimg);
+
+        const int white_margin = 2;
+        const int finder_width = 7;
+
+        const int timing_pos = white_margin + 6;
+        int i;
+
+        // Horizontal Check
+        // (1) White margin(Left)
+        for(i = 0; i < white_margin ; i++ )
+        {
+            CV_Assert( qrimg.at<uint8_t>(i, timing_pos) == 255 ) ;
```

Comment:
> CV_Assert

Please use GoogleTest checks instead:

- `ASSERT_EQ(expected_value, actual_value)` << "i=" << i;

Please note about the order of parameters:
`ASSERT_EQ((uint8_t)255, qrimg.at<uint8_t>(i, timing_pos)) << "i=" << i;`

BTW, `ASSERT_` could be replaced by `EXPECT_` if there number of error messages is limited and small (we should not trash logs in case of failure).